New version crash on launch

**Platform**
Android 10

**Version**
Google play store 4.1.16

**Describe the bug**
Cannot launch the game and crash everytime

**To Reproduce**
Steps to reproduce the behavior:
1. Update to newest version from gps
2. Open game
3. Wait a while
4. Crash

**Expected behavior**
Not crash

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Was opening unciv while updating it (cleared the old version out of background then open the new version)

https://user-images.githubusercontent.com/107736312/177994702-5810e2c0-5708-4d42-bd57-8d69e54dd458.mp4

`com.badlogic.gdx.utils.SerializationException: Error reading file: MultiplayerGames/2v2 w
	at com.badlogic.gdx.utils.Json.fromJson(Json.java:803)
	at com.unciv.logic.UncivFiles.loadGamePreviewFromFile(UncivFiles.kt:233)
	at com.unciv.logic.multiplayer.OnlineMultiplayer.addGame$default(OnlineMultiplayer.kt:158)
	at com.unciv.logic.multiplayer.OnlineMultiplayer.updateSavesFromFiles(OnlineMultiplayer.kt:113)
	at com.unciv.logic.multiplayer.OnlineMultiplayer.access$updateSavesFromFiles(OnlineMultiplayer.kt:42)
	at com.unciv.logic.multiplayer.OnlineMultiplayer$updateSavesFromFiles$1.invokeSuspend(Unknown Source:14)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at com.unciv.utils.concurrency.CrashHandlingDispatcher$dispatch$1.invoke(Concurrency.kt:164)
	at com.unciv.utils.concurrency.CrashHandlingDispatcher$dispatch$1.invoke(Concurrency.kt:164)
	at com.unciv.ui.crashhandling.CrashHandlingExtensionsKt$wrapCrashHandling$1.invoke(CrashHandlingExtensions.kt:17)
	at com.unciv.ui.crashhandling.CrashHandlingExtensionsKt$wrapCrashHandlingUnit$1.invoke(CrashHandlingExtensions.kt:33)
	at com.unciv.ui.crashhandling.CrashHandlingExtensionsKt$wrapCrashHandlingUnit$1.invoke(CrashHandlingExtensions.kt:33)
	at com.unciv.utils.concurrency.CrashHandlingDispatcher.dispatch$lambda-0(Concurrency.kt:164)
	at com.unciv.utils.concurrency.CrashHandlingDispatcher.$r8$lambda$wZEPzfc3ap1lx3h-gfSjlNGDZEw(Unknown Source:0)
	at com.unciv.utils.concurrency.CrashHandlingDispatcher$$ExternalSyntheticLambda0.run(Unknown Source:2)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:919)
Caused by: com.badlogic.gdx.utils.SerializationException: Error reading file: MultiplayerGames/2v2 w
	at com.badlogic.gdx.utils.JsonReader.parse(JsonReader.java:79)
	at com.badlogic.gdx.utils.Json.fromJson(Json.java:801)
	... 18 more
Caused by: com.badlogic.gdx.utils.GdxRuntimeException: Cannot open a stream to a directory: MultiplayerGames/2v2 w (External)
	at com.badlogic.gdx.files.FileHandle.read(FileHandle.java:149)
	at com.badlogic.gdx.backends.android.AndroidFileHandle.read(AndroidFileHandle.java:86)
	at com.badlogic.gdx.files.FileHandle.reader(FileHandle.java:169)
	at com.badlogic.gdx.utils.JsonReader.parse(JsonReader.java:77)
	... 19 more
Caused by: java.io.FileNotFoundException: /storage/emulated/0/Android/data/com.unciv.app/files/MultiplayerGames/2v2 w: open failed: EISDIR (Is a directory)
	at libcore.io.IoBridge.open(IoBridge.java:496)
	at java.io.FileInputStream.<init>(FileInputStream.java:159)
	at com.badlogic.gdx.files.FileHandle.read(FileHandle.java:146)
	... 22 more
Caused by: android.system.ErrnoException: open failed: EISDIR (Is a directory)
	at libcore.io.IoBridge.open(IoBridge.java:486)
	... 24 more
`

--- a/core/src/com/unciv/logic/UncivFiles.kt
+++ b/core/src/com/unciv/logic/UncivFiles.kt
@@ -245,7 +245,7 @@ class UncivFiles(
     /**
      * GDX JSON deserialization does not throw when the file is empty, it just returns `null`.
      *
-     * Since we work with non-nullable types, we convert this to an actual exception here ot have a nice exception message instead of a NPE.
+     * Since we work with non-nullable types, we convert this to an actual exception here to have a nice exception message instead of a NPE.
      */
     private fun emptyFile(gameFile: FileHandle): SerializationException {
         return SerializationException("The file for the game ${gameFile.name()} is empty")

--- a/core/src/com/unciv/logic/UncivFiles.kt
+++ b/core/src/com/unciv/logic/UncivFiles.kt
@@ -4,6 +4,7 @@ import com.badlogic.gdx.Files
 import com.badlogic.gdx.Gdx
 import com.badlogic.gdx.files.FileHandle
 import com.badlogic.gdx.utils.JsonReader
+import com.badlogic.gdx.utils.SerializationException
 import com.unciv.UncivGame
 import com.unciv.json.fromJsonFile
 import com.unciv.json.json
@@ -223,14 +224,31 @@ class UncivFiles(
             loadGameFromFile(getSave(gameName))
 
     fun loadGameFromFile(gameFile: FileHandle): GameInfo {
-        return gameInfoFromString(gameFile.readString())
+        val gameData = gameFile.readString()
+        if (gameData.isNullOrBlank()) {
+            throw emptyFile(gameFile)
+        }
+        return gameInfoFromString(gameData)
     }
 
     fun loadGamePreviewByName(gameName: String) =
             loadGamePreviewFromFile(getMultiplayerSave(gameName))
 
     fun loadGamePreviewFromFile(gameFile: FileHandle): GameInfoPreview {
-        return json().fromJson(GameInfoPreview::class.java, gameFile)
+        val preview = json().fromJson(GameInfoPreview::class.java, gameFile)
+        if (preview == null) {
+            throw emptyFile(gameFile)
+        }
+        return preview
+    }
+
+    /**
+     * GDX JSON deserialization does not throw when the file is empty, it just returns `null`.
+     *
+     * Since we work with non-nullable types, we convert this to an actual exception here ot have a nice exception message instead of a NPE.
+     */
+    private fun emptyFile(gameFile: FileHandle): SerializationException {
+        return SerializationException("The file for the game ${gameFile.name()} is empty")
     }
 
     class CustomLoadResult<T>(

--- a/core/src/com/unciv/logic/multiplayer/OnlineMultiplayer.kt
+++ b/core/src/com/unciv/logic/multiplayer/OnlineMultiplayer.kt
@@ -21,7 +21,6 @@ import kotlinx.coroutines.coroutineScope
 import kotlinx.coroutines.delay
 import kotlinx.coroutines.flow.flow
 import kotlinx.coroutines.flow.launchIn
-import java.io.FileNotFoundException
 import java.time.Duration
 import java.time.Instant
 import java.util.*
@@ -155,9 +154,9 @@ class OnlineMultiplayer {
         return addGame(fileHandle, preview)
     }
 
-    private suspend fun addGame(fileHandle: FileHandle, preview: GameInfoPreview = files.loadGamePreviewFromFile(fileHandle)) {
-        debug("Adding game %s", preview.gameId)
-        val game = OnlineMultiplayerGame(fileHandle, preview, Instant.now())
+    private suspend fun addGame(fileHandle: FileHandle, preview: GameInfoPreview? = null) {
+        debug("Adding game %s", fileHandle.name())
+        val game = OnlineMultiplayerGame(fileHandle, preview, if (preview != null) Instant.now() else null)
         savedGames[fileHandle] = game
         withGLContext {
             EventBus.send(MultiplayerGameAdded(game.name))

